File: ocaml/chain.ml
(* OCaml transitive chain via 3 nested modules. *)

module Origin = struct
  let origin_value = "T1.origin"                         (* T1.origin.def *)
end

module Middle = struct
  let middle_value = Origin.origin_value                 (* T1.middle.reexport *)
end

module Deep = struct
  let value_alias = Middle.middle_value                  (* T1.deep.reexport *)
end

let run () = print_endline ("transitive: " ^ Deep.value_alias)

File: zig/chain.zig
// T1 transitive chain via 3 files — Zig needs explicit @import at each hop.

const origin = @import("chain_origin.zig");
const middle = @import("chain_middle.zig");
const deep   = @import("chain_deep.zig");

pub fn run() void {
    _ = origin.ORIGIN_VALUE;
    _ = middle.MIDDLE_VALUE;
    const alias: []const u8 = deep.VALUE_ALIAS;    // T1.consumer.read
    @import("std").debug.print("transitive: {s}\n", .{alias});
}
